sentry/src/main/java/io/sentry/Attachment.java

```diff
@@ -230,4 +284,31 @@ boolean isAddToTransactions() {
   public static @NotNull Attachment fromScreenshot(final byte[] screenshotBytes) {
     return new Attachment(screenshotBytes, "screenshot.png", "image/png", false);
   }
+
+  /**
+   * Creates a new View Hierarchy Attachment
+   *
+   * @param viewHierarchy the View Hierarchy
+   * @return the Attachment
+   */
+  public static @Nullable Attachment fromViewHierarchy(final ViewHierarchy viewHierarchy) {
```

Comment:
@romtsn I really don't like this part
* this will do the serialization on the main thread
* it's accessing `Sentry.getCurrentHub().getOptions().getSerializer()` which makes it hard to test

Now I'm wondering: Would it make sense to refactor the `Attachment` class and provide an option for having serialize-able payload?
e.g.
````
abstract class Attachment {}

class RawAttachment extends Attachment {
  private final byte[] bytes;
}

class JsonAttachment extends Attachment {
  private final JsonSerializable serializable;
}
````

Reply:
re. options - I think we can just pass them as a parameter from the event processor, right? then it'd be testable.

re. serialization on main thread - I think we can offload it to another thread, but should still block and wait for it, because we're doing it only for errored events and the app is gonna crash anyway? That's what we do for screenshots as well. But I'm wondering how other SDKs did that? @marandaneto @brustolin 

Imo we should block here, because the app may crash, and if we don't wait for it, then we'll lose the attachment

Reply:
@markushi I do the serialization on the background, you can do the same on Android, attachments already have a callback that is only called during the event sending (and reads the bytes async).
You can check the Dart PR or the Android code.

Reply:
I moved the serialization part to the`ViewHierarchyProcessor`, looking way better now in terms of testability.
re. serialization on main thread: You're right, view hierarchies are only generated when the app crashes, thus it shouldn't be an issue. (On a another note: The Android -> Native scope sync is performing serialization on the main thread as well, maybe that's something we should fix).

Reply:
For cocoa, if it's a crash we serialize it in the crash thread, which is not the main. And if is an error capture we serialize in the main thread otherwise there is a bunch of warnings that we are calling UI methods in a background thread.